File: frontend/app/src/api/userImage.ts
import request from '../utils/request'

export type UserImage = {
  id: number
  user_id: number
  title?: string | null
  image_url: string
  created_at: string
}

export function listMyUserImages() {
  return request.get<UserImage[], UserImage[]>('/user-images/me')
}

File: backend/app/models/user_image.py
from sqlalchemy import Column, DateTime, ForeignKey, Integer, String
from datetime import datetime

from app.db.database import Base


class UserImage(Base):
    __tablename__ = "user_images"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False, index=True)
    title = Column(String(200), nullable=True)
    image_url = Column(String(2048), nullable=False)
    created_at = Column(DateTime, default=datetime.now)
